lldb/source/Plugins/SymbolFile/CTF/SymbolFileCTF.cpp

```diff
@@ -1032,11 +1032,17 @@ void SymbolFileCTF::FindTypes(const lldb_private::TypeQuery &match,
 
   ConstString name = match.GetTypeBasename();
   for (TypeSP type_sp : GetTypeList().Types()) {
-    if (type_sp && type_sp->GetName() == name) {
-      results.InsertUnique(type_sp);
-      if (results.Done(match))
-        return;
-    }
+    if (!type_sp)
+      continue;
+    auto type_name =
+        match.GetSearchByMangledName()
+            ? type_sp->GetForwardCompilerType().GetMangledTypeName()
+            : type_sp->GetName();
+    if (type_name != name)
+      continue;
+    results.InsertUnique(type_sp);
+    if (results.Done(match))
+      return;
```

Comment:
Also, do we need to implement this for new lookup option for `CTF` and `PDB`? Given it will probably be always unused (e.g., CTF is only used with `C` IIUC), maybe lets limit this change to just DWARF for now?

Reply:
Sure, I implemented there for completeness.